game-core/src/main/java/games/strategy/triplea/attachments/UnitAttachment.java

```diff
@@ -629,13 +633,10 @@ private void setIsSub(final String s) {
   }
 
   private void setIsSub(final Boolean s) {
-    setCanEvade(s);
-    setIsFirstStrike(s);
-    setCanMoveThroughEnemies(s && Properties.getSubmersibleSubs(getData()));
-    setCanBeMovedThroughByEnemies(s && Properties.getIgnoreSubInMovement(getData()));
+    isSub = true;
     if (s) {
       canNotTarget = null;
-      canNotBeTargetedBy = Properties.getAirAttackSubRestricted(getData()) ? null : new HashSet<>();
+      canNotBeTargetedBy = null;
```

Comment:
Codacy found an issue: [Assigning an Object to null is a code smell.  Consider refactoring.](https://app.codacy.com/gh/triplea-game/triplea/pullRequest?prid=5897929)